core/auth/src/main/java/software/amazon/awssdk/auth/signer/Aws4UnsignedPayloadSigner.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.auth.signer;
+
+import static software.amazon.awssdk.auth.signer.internal.SignerConstant.X_AMZ_CONTENT_SHA256;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.auth.signer.params.Aws4SignerParams;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+
+/**
+ * Exactly the same as {@link Aws4Signer} except if the request is being sent
+ * over HTTPS, then it returns the string <code>UNSIGNED-PAYLOAD</code> as the
+ * content SHA-256 so services that support it can avoid needing to calculate
+ * the value when authorizing the request.
+ * <p>
+ * Payloads are still signed for requests over HTTP to preserve the request
```

Comment:
Where does this happen?

Reply:
For http requests, [this](https://github.com/aws/aws-sdk-java-v2/blob/e8aee9db60fefad5b13ae16ae5540820f7037e72/core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AbstractAws4Signer.java#L146-L159) super class method is invoked